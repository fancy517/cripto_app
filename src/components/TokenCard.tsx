import { Text, View } from "react-native";
import React, { Component } from "react";
import Token from "../types/Token";

type TokenProps = {
  token: Token;
};

export class TokenCard extends Component<TokenProps> {
  state = {};
  render() {
    return (
      <View>
        <Text>{this.props.token.name}</Text>
      </View>
    );
  }
}

export default TokenCard;
