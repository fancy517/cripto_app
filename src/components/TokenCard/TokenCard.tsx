import Token from "../../types/Token";
import { Card } from "react-native-paper";
import { Image, StyleSheet } from "react-native";
import RowOfText from "../RowOfText";

type TokenProps = {
  token: Token;
  navigate: (name: string, {}) => void;
};

export default function TokenCard({ token, navigate }: TokenProps) {
  return (
    <Card
      style={style.cardContainer}
      onPress={() => {
        navigate("TicketDetails", { ticketId: token.id });
      }}
      mode="contained"
    >
      <Card.Title
        title={token.name}
        titleVariant="headlineMedium"
        left={() => (
          <Image
            style={{ height: 40, width: 40 }}
            source={{
              uri: `https://assets.coincap.io/assets/icons/${token.symbol.toLowerCase()}@2x.png`,
            }}
          />
        )}
      />
      <Card.Content>
        <RowOfText name="Price:" value={`$${token.price_usd} USD`} />
        <RowOfText
          name="Last 24 Hours:"
          value={`${token.percent_change_24h}%`}
        />
      </Card.Content>
    </Card>
  );
}

const style = StyleSheet.create({
  cardContainer: {
    margin: 5,
  },
  title: {
    marginTop: 0,
  },
});
