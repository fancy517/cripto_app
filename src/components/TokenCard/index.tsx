import Token from "../../types/Token";
import { Card } from "react-native-paper";
import { Image, StyleSheet } from "react-native";
import { useNavigation } from "@react-navigation/native";
import { RootStackParamsList } from "../../types/StackParams";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import RowOfText from "../RowOfText";

type TokenProps = {
  token: Token;
};

type TicketDetailsProps = NativeStackNavigationProp<
  RootStackParamsList,
  "TicketDetails"
>;

export default function TokenCard({ token }: TokenProps) {
  const navigation = useNavigation<TicketDetailsProps>();
  return (
    <Card
      style={style.cardContainer}
      onPress={() => {
        navigation.navigate("TicketDetails", { ticketId: token.id });
      }}
      mode="contained"
    >
      <Card.Title
        title={token.name}
        titleVariant="headlineMedium"
        left={() => (
          <Image
            style={{ height: 40, width: 40 }}
            source={{
              uri: `https://assets.coincap.io/assets/icons/${token.symbol.toLowerCase()}@2x.png`,
            }}
          />
        )}
      />
      <Card.Content>
        <RowOfText name="Price:" value={`$${token.price_usd} USD`} />
        <RowOfText
          name="Last 24 Hours:"
          value={`${token.percent_change_24h}%`}
        />
      </Card.Content>
    </Card>
  );
}

const style = StyleSheet.create({
  cardContainer: {
    margin: 5,
  },
  title: {
    marginTop: 0,
  },
});
