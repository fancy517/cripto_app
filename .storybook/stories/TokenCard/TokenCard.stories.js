import { TokenCardMeta } from './TokenCardMeta';

export default TokenCardMeta;

export const Basic = {};

export const Alternative = {

    args: {
        token: {
            "id": "90",
            "symbol": "BTC",
            "name": "Bitcoin",
            "nameid": "bitcoin",
            "rank": 1,
            "price_usd": "26132.55",
            "percent_change_24h": "-0.43",
            "percent_change_1h": "-0.20",
            "percent_change_7d": "-2.58",
            "price_btc": "1.00",
            "market_cap_usd": "508825486921.92",
            "volume24": 6280637402.078317,
            "volume24a": 7979003931.446185,
            "csupply": "19470946.00",
            "tsupply": "19470946",
            "msupply": "21000000"
        }

    },
};
