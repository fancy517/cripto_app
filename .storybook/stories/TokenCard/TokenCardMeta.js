import React from 'react';
import { View } from 'react-native';
import TokenCard from '../../../src/components/TokenCard/TokenCard';
import { NavigationContainer } from '@react-navigation/native';

export const TokenCardMeta = {
    title: 'TokenCard',
    component: TokenCard,
    argTypes: {
        onPress: { action: 'pressed the button' },
    },
    args: {
        token: { "id": "90", "symbol": "BTC", "name": "Bitcoin", "nameid": "bitcoin", "rank": 1, "price_usd": "26132.55", "percent_change_24h": "-0.43", "percent_change_1h": "-0.20", "percent_change_7d": "-2.58", "price_btc": "1.00", "market_cap_usd": "508825486921.92", "volume24": 6280637402.078317, "volume24a": 7979003931.446185, "csupply": "19470946.00", "tsupply": "19470946", "msupply": "21000000" },
        navigate: () => { }
    },
    decorators: [
        (Story) => (
            <View style={{ justifyContent: 'center', flex: 1, width: '100%' }}>
                <NavigationContainer>
                    <Story />
                </NavigationContainer>
            </View>
        ),
    ],
};
