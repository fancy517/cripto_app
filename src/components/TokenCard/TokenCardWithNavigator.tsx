import TokenCard from "./TokenCard";
import Token from "../../types/Token";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { RootStackParamsList } from "../../types/StackParams";
import { useNavigation } from "@react-navigation/native";

type TokenProps = {
  token: Token;
};

type TicketDetailsProps = NativeStackNavigationProp<
  RootStackParamsList,
  "TicketDetails"
>;

const TokenCardWithNavigator = ({ token }: TokenProps) => {
  const navigation = useNavigation<TicketDetailsProps>();
  return <TokenCard token={token} navigate={navigation.navigate} />;
};

export default TokenCardWithNavigator;
