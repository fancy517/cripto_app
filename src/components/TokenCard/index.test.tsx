import React from "react";
import { create, act } from "react-test-renderer";
import TokenCard from "./index";
import { NavigationContainer } from "@react-navigation/native";
// import Token from "../../types/Token";
jest.setTimeout(1000);
const token = {
  csupply: "",
  id: "1",
  market_cap_usd: "",
  msupply: "",
  name: "2",
  nameid: "",
  percent_change_1h: "",
  percent_change_24h: "",
  percent_change_7d: "",
  price_btc: "",
  price_usd: "",
  rank: 1,
  symbol: "",
  tsupply: "",
  volume24: 1,
  volume24a: 1,
};

// const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
jest.useFakeTimers();

describe("TokenCard Tests", () => {
  it("matches snapshot", async () => {
    // await sleep(2000);
    let jsonElement = create(
      <NavigationContainer>
        <TokenCard token={token} />
      </NavigationContainer>
    ).toJSON();
    act(() => {
      expect(jsonElement).toMatchSnapshot();
    });
  });
});
