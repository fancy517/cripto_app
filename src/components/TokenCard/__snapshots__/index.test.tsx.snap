// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenCard Tests matches snapshot 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "rgba(231, 224, 236, 1)",
      "borderRadius": 12,
      "margin": 5,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(231, 224, 236, 1)",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="card-container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingLeft": 16,
            },
            {
              "minHeight": 72,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "height": 40,
                "justifyContent": "center",
                "marginRight": 16,
                "width": 40,
              },
              undefined,
            ]
          }
        >
          <Image
            source={
              {
                "uri": "https://assets.coincap.io/assets/icons/@2x.png",
              }
            }
            style={
              {
                "height": 40,
                "width": 40,
              }
            }
          />
        </View>
        <View
          style={
            [
              {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 28,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 36,
                  },
                  [
                    {
                      "minHeight": 30,
                      "paddingRight": 16,
                    },
                    {
                      "marginBottom": 2,
                    },
                    undefined,
                  ],
                ],
              ]
            }
          >
            2
          </Text>
        </View>
        <View />
      </View>
      <View
        borderRadiusStyles={{}}
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "paddingBottom": 16,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
          >
            Price:
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
          >
            $ USD
          </Text>
        </View>
        <View
          style={
            {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
          >
            Last 24 Hours:
          </Text>
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0.15,
                    "lineHeight": 24,
                  },
                  undefined,
                ],
              ]
            }
          >
            %
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
